### PR TITLE
Add helper methods to projection

### PR DESCRIPTION
**Pull request by StefanoPetrilli**
In https://github.com/godotengine/godot/pull/77275 @AThousandShips  suggested adding some helper methods to `Projection`.
This PR adds those helper methods and updates the tests to use `is_equal_approx `.
This PR is currently based on https://github.com/godotengine/godot/pull/77275.

### CODE REVIEW

**Comment by AThousandShips** [comment]
The `is_equal_approx` and `is_finite` should probably be exposed to GDScript as well to make them useful

**Review by AThousandShips — APPROVED** [review]
LGTM

**Review by AThousandShips — DISMISSED** [review]
LGTM still, avoid re-requesting reviews unless something significant has changed (not to mention when nothing has changed)

**Comment by AThousandShips** [comment]
You need to update the documentation, and add descriptions for the new methods

File: doc/classes/Projection.xml [hunk 1]
```diff
@@ -240,6 +240,19 @@
 				Returns a [Projection] that performs the inverse of this [Projection]'s projective transformation.
 			</description>
 		</method>
+		<method name="is_equal_approx" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="to" type="Projection" />
+			<description>
+				Returns [code]true[/code] if this [Projection] and the given [Projection] are equal within a small tolerance.
```
Comment (AThousandShips) [inline doc/classes/Projection.xml:247]:
```suggestion
				Returns [code]true[/code] if this [Projection] and [param to] are approximately equal, by calling [method @GlobalScope.is_equal_approx] on each component.
```
To match the style in `AABB`, `Transform2D`, etc.

Comment (StefanoPetrilli) [reply doc/classes/Projection.xml]: It should be all right now

File: tests/core/math/test_projection.h [hunk 2]
```diff
@@ -0,0 +1,807 @@
...
+}
+
+TEST_CASE("[Projection] get_pixels_per_meter") {
+	int expected = 55;
+	int actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+						   .get_pixels_per_meter(64);
```
Comment (clayjohn) [inline tests/core/math/test_projection.h:600]:
```suggestion
						 .get_pixels_per_meter(64);
```

Looks like clang format isn't happy with this whitespace. I just copy-pasted the suggestion from the failing CI

**Pull request closed without merging**